### PR TITLE
Simplify errors

### PR DESCRIPTION
**Pull request by campoy**
Some refactoring, have a look @crhym3

### CODE REVIEW

File: endpoints/errors.go [hunk 1]
```diff
@@ -69,39 +55,39 @@ func NewAPIError(name string, msg string, code int) error {
 	return &APIError{Name: name, Msg: msg, Code: code}
 }
 
-// Errorf creates a new APIError for custom error given a format string and arguments.
-func Errorf(code int, name string, format string, args ...interface{}) error {
-	return &APIError{Name: name, Msg: fmt.Sprintf(format, args...), Code: code}
+// errorf creates a new APIError given its status code, a format string and its arguments.
+func errorf(code int, format string, args ...interface{}) error {
```
Comment (crhym3) [inline endpoints/errors.go]: So, you're basically saying if needs a custom error, they can just do `&APIError{...}`.

**Comment by crhym3** [comment]
Thanks for refactoring `errorNames`! I never liked that slice at all.
LGTM

**Comment by crhym3** [comment]
Wait...

File: endpoints/errors.go [hunk 2]
```diff
@@ -14,41 +14,27 @@ var (
...
 
 	// InternalServerError is default error with http.StatusInternalServerError (500)
-	InternalServerError = NewInternalServerError(errorNames[0])
+	InternalServerError = NewInternalServerError("")
 	// BadRequestError is default error with http.StatusBadRequest (400)
-	BadRequestError = NewBadRequestError(errorNames[1])
+	BadRequestError = NewBadRequestError("")
 	// UnauthorizedError is default error with http.StatusUnauthorized (401)
-	UnauthorizedError = NewUnauthorizedError(errorNames[2])
+	UnauthorizedError = NewUnauthorizedError("")
```
Comment (crhym3) [inline endpoints/errors.go]: So, what if I `return endpoints.UnauthorizedError` from a method. It's `errorResponse` will have zero `Msg` field?

File: endpoints/errors.go [hunk 3]
```diff
@@ -14,41 +14,27 @@ var (
...
-		http.StatusForbidden,
-		http.StatusNotFound,
-		http.StatusConflict,
+	errorNames = map[int]string{
+		http.StatusInternalServerError: "Internal Server Error",
+		http.StatusBadRequest:          "Bad Request",
+		http.StatusUnauthorized:        "Unauthorized",
+		http.StatusForbidden:           "Forbidden",
+		http.StatusNotFound:            "Not Found",
+		http.StatusConflict:            "Conflict",
```
Comment (crhym3) [inline endpoints/errors.go]: Maybe we could use `http.StatusText(...)` here or in `New*Error()` funcs?

File: endpoints/errors.go (hunk 2 above)
Comment (campoy) [inline endpoints/errors.go]: That was already the case, actually I'm not sure that we should provide these error values.

File: endpoints/errors.go (hunk 1 above)
Comment (campoy) [inline endpoints/errors.go]: Basically, yes.

File: endpoints/errors.go (hunk 3 above)
Comment (campoy) [inline endpoints/errors.go]: done

**Comment by adg** [comment]
LGTM

**Pull request merged**